lib/auth0/appEndpointHandler.js

```diff
@@ -6,11 +6,15 @@ module.exports = function (webtask, options, ctx, req, res, routingInfo) {
     function authenticate() {
         var token = options.getAccessToken(ctx, req);
         if (!token) {
+            var stateInfo = {
+                appPath: routingInfo.appPath,
+                query: routingInfo.query
+            };
             return options.loginError({
                 code: 401,
                 message: 'Unauthorized.',
                 error: 'Missing access token.',
-                redirect: routingInfo.baseUrl + '/login'
+                redirect: routingInfo.baseUrl + '/login?state=' + encodeURIComponent(JSON.stringify(stateInfo))
```

Comment:
I like that approach, as the state information only exists between the user agent and the webtask code, and isn't shared with other parties, including the authentication service.